pkg/controller/embercsi/common.go

```diff
@@ -263,7 +267,8 @@ func generateVolumeMounts(ecsi *embercsiv1alpha1.EmberCSI, csiDriverMode string)
 	}
 
 	// Check to see if the volume driver is LVM
-	if len(ecsi.Spec.Config.EnvVars.X_CSI_BACKEND_CONFIG) > 0 && strings.Contains(strings.ToLower(ecsi.Spec.Config.EnvVars.X_CSI_BACKEND_CONFIG), "lvmvolume") {
+	X_CSI_BACKEND_CONFIG := interfaceToString(ecsi.Spec.Config.EnvVars.X_CSI_BACKEND_CONFIG)
+	if len(X_CSI_BACKEND_CONFIG) > 0 && strings.Contains(strings.ToLower(X_CSI_BACKEND_CONFIG), "lvmvolume") {
```

Comment:
Isn't this kind of broad? This should be checking just the "driver" part, right? Not the whole json blob.

Reply:
Yes, fully agree, and it's on my list for future patches :)